### PR TITLE
[MB-17020] Update transportation office foreign key constraint to ON DELETE CASCADE

### PR DESCRIPTION
**Pull request by melissathai**
## Summary

We would like to remove the 48 transportation offices listed in [this spreadsheet](https://docs.google.com/spreadsheets/d/1dBBNKveq4eAslln3qQCF3UbEvM5azuBj8hcxGuEr_WY/edit#gid=15177220). However, before doing so, the `office_phone_lines.transportation_office_id` and `office_emails.transportation_office_id` foreign key constraints must be updated to `ON DELETE CASCADE` to ensure that when a TO is deleted, so is its corresponding office phone lines and office emails (note that new phone lines and emails have already been created for the new transportation offices).

## Verification Steps for the Author

- [x] Ran database migration locally and manually deleted a TO with a phone line and email

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.

### CODE REVIEW

File: migrations/app/schema/20230915055143_update_transportation_office_foreign_key_constraints.up.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+-- Drop the existing foreign key constraint
+ALTER TABLE office_emails
+DROP CONSTRAINT office_emails_transportation_office_id_fkey;
+
+-- Add the ON DELETE CASCADE constraint
+ALTER TABLE office_emails
+ADD CONSTRAINT office_emails_transportation_office_id_fkey
+FOREIGN KEY (transportation_office_id)
+REFERENCES transportation_offices(id)
+ON DELETE CASCADE;
```
Comment (kctruss) [inline migrations/app/schema/20230915055143_update_transportation_office_foreign_key_constraints.up.sql:30]: Do we want to keep these `ON DELETE CASCADE` constraints after the data migration is completed, or should we plan to follow up with another migration to reset the constraints?

Comment (melissathai) [reply migrations/app/schema/20230915055143_update_transportation_office_foreign_key_constraints.up.sql:30]: I think we'd like to keep these constraints as they ensure data integrity. I can't think of a case where we delete a transportation office and want to keep its address.

Comment (kctruss) [reply migrations/app/schema/20230915055143_update_transportation_office_foreign_key_constraints.up.sql:30]: Makes sense to me!

**Review by kctruss — APPROVED** [review]
🌈 looks good to me!

**Pull request merged**